scripts/generateCredentials.js

```diff
@@ -0,0 +1,145 @@
+/*!
+Copyright 2019 Raising the Floor US
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+
+As part of GPII-3714, we need a script to generate the credentials
+for both the client and the server in order for the instances of GPII to be
+authenticated when performing read and save actions against the cloud-based flowManager.
+
+These credentials consist in:
+
+1. The secret file - what goes in the client side
+
+{
+  "site": "example.gpii.net",
+  "clientCredentials": {
+    "client_id": "UUID-pilot-computer",
+    "client_secret": "UUID-pilot-computer-secret"
+  }
+}
+
+2. The credentials - what goes into the database
+
+2.1 The client credentials
+
+{
+    "_id": "UUID-clientCredentials",
+    "type": "clientCredential",
+    "schemaVersion": "0.1",
+    "clientId": "UUID-gpiiAppInstallationClient",
+    "oauth2ClientId": "UUID-pilot-computer",
+    "oauth2ClientSecret": "UUID-pilot-computer-secret",
+    "revoked": false,
+    "revokedReason": null,
+    "timestampCreated": "2017-11-21T18:11:22.101Z",
+    "timestampRevoked": null
+}
+
+2.2 The GPII App Installation Client
+
+{
+    "_id": "UUID-gpiiAppInstallationClient",
+    "type": "gpiiAppInstallationClient",
+    "schemaVersion": "0.1",
+    "name": "Pilot Computers",
+    "computerType": "public",
+    "timestampCreated": "2017-11-21T18:11:22.101Z",
+    "timestampUpdated": null
+}
+
+The options must be passed as environmental variables, they are:
+
+* GPII_CREDENTIALS_NAME [String] [required]: The site name
+* GPII_CREDENTIALS_SITE [String] [required]: The site domain
+
+Example usage:
+
+GPII_CREDENTIALS_NAME="GPII Testers" GPII_CREDENTIALS_SITE=testers.gpii.net node generateCredentials.js
+
+After running the script, you can find the files in the folder "testers.gpii.net-credentials".
+The script generates two files containing:
+1. secret.txt: The file to be provided when building the "Secret Blower" installer
+2. couchDBData.json: The JSON-format documents that go into the production
+database and ready to be inserted using an HTTP POST request like this:
+
+* curl -d @couchDBData.json -H "Content-type: application/json" -X POST http://localhost:25984/gpii/_bulk_docs
```

Comment:
I guess the plan is to run something similar to this workflow in production (and stg)?

If so, we will want a note in gpii-infra; the [one you wrote for seeding preferences for users](https://github.com/gpii-ops/gpii-infra/blob/master/gcp/README.md#manual-process-generating-db-data-of-user-preferences-and-load-it-into-a-clusters-preferences-server) is a good example.